File: minMax.py
# Author: Naveen US
# Titile: Find min max of an array.

# Time complexity: O(n)
# Space complexity: O(1)
# Did the code run on leetcode: Yes

# Disappeared elements are found in place.  

def min_and_max(array):
    # if array doesn't have values. 
    if len(array)==0:
        return [None, None]
    else:
        #initialize and loop.
        minimum = array[0]
        maximum = array[0]
        for i in range(1, len(array)):
            if array[i]>maximum:
                maximum = array[i]
            if array[i]<minimum:
                minimum = array[i]
        return [minimum, maximum]

File: min-max.js
// Time Complexity: O(n)
// Space Complexity: O(1)
// Did this code successfully run on Leetcode: not found on leetcode
// Any problem you faced while coding this: No

// [1, 3, 5, 2, 3, 7, 1]

const findMinMax = (nums) => {
  if (nums == null || nums.length == 0) return [];

  let min = Infinity,
      max = Number.NEGATIVE_INFINITY;

  for (const num of nums) {
    min = Math.min(num, min);
    max = Math.max(num, max);
  }
  return [min, max];
}
